graphics/nxterm/nxterm_kbdin.c

```diff
@@ -128,6 +129,7 @@ ssize_t nxterm_read(FAR struct file *filep, FAR char *buffer, size_t len)
            * to wake us up.
            */
 
+          flags = enter_critical_section();
```

Comment:
From the change description: "in smp sched_lock is not enough to prevent preemption", but I'm missing what is the state of the problem and how the critical section helps here.

Reply:
This modification is very similar to the implementation of pthread_cond_wait
 https://github.com/apache/nuttx/blob/ad6b12acf56aa1da585ccf23cde1362204ba7c0e/sched/pthread/pthread_condwait.c#L96C2-L96C2